nnstreamer_example/custom_example_tensorrt/nnstreamer_customfilter_example_tensorrt_reshape.cc

```diff
@@ -0,0 +1,484 @@
+/**
+ * NNStreamer TensorRT Custom Filter Example: Reshape
+ *
+ * Copyright (C) 2020 Dongju Chae <dongju.chae@samsung.com>
+ *
+ * SPDX-License-Identifier: LGPL-2.1-only
+ *
+ * @file  nnstreamer_customfilter_example_tensorrt_reshape.cc
+ * @date  14 Sep 2020
+ * @brief  TensorRT Custom NNStreamer Filter Example: Reshape
+ * @author  Dongju Chae <dongju.chae@samsung.com>
+ * @bug  No known bugs
+ *
+ * Reference code: sampleDynamicReshape.cpp in TensorRT samples
+ *
+ * This example reshapes the inputs to the given dimensions.
+ * The custom property is to be given as, "custom=D1:D2:D3"
+ * E.g., custom=224:224:3
+ *
+ */
+
+#include <glib.h>
+#include <string.h>
+#include <tensor_filter_custom.h>
+#include <nnstreamer_plugin_api.h>
+#include <nnstreamer_log.h>
+
+#include <iostream>
+#include <memory>
+#include <vector>
+
+#include <NvInfer.h>
+#include <NvInferRuntimeCommon.h>
+#include <cuda_runtime_api.h>
+
+#ifdef USE_UNIFIED_MEMORY
+#define CUDA_ALLOC cudaMallocManaged
+#else
+#define CUDA_ALLOC cudaMalloc
+#endif
+
+#define CUDA_FREE cudaFree
+
+using Severity = nvinfer1::ILogger::Severity;
+
+/** @brief a global object of ILogger */
+class Logger : public nvinfer1::ILogger
+{
+	void log(Severity severity, const char* msg) override
+	{
+    switch (severity) {
+      case Severity::kWARNING:
+        ml_logw ("%s", msg);
+        break;
+      case Severity::kINFO:
+        ml_logi ("%s", msg);
+        break;
+      case Severity::kVERBOSE:
+        ml_logd ("%s", msg);
+        break;
+      default:
+        ml_loge ("%s", msg);
+        break;
+    }
+	}
+} gLogger;
+
+/** @brief unique ptr deleter */
+struct InferDeleter
+{
+  template <typename T>
+  void operator()(T* obj) const
+  {
+    if (obj)
+      obj->destroy();
+  }
+};
+
+/**
+ * @brief class definition for TensorRT custom filter
+ */
+class CustomTensorRT
+{
```

Comment:
If you are going to use C++ classes, use C++ custom filter API: https://github.com/nnstreamer/nnstreamer/blob/master/ext/nnstreamer/tensor_filter/tensor_filter_cpp.hh

Reply:
Got it

Reply:
Well.. It seems that `destoryNotify` is not implemented in C++ custom filter.

Reply:
Oh... right.. I'll add an issue regarding this.